src/cargo/util/errors.rs

```diff
@@ -286,6 +286,37 @@ impl From<Box<CargoError>> for CliError {
 }
 
 // =============================================================================
+// NetworkError trait
+
+pub trait NetworkError: CargoError {
+    fn maybe_spurious(&self) -> bool;
+}
+
+impl NetworkError for git2::Error {
+    fn maybe_spurious(&self) -> bool {
+        match self.class() {
+            git2::ErrorClass::Net |
+            git2::ErrorClass::Os => true,
+            _ => false
+        }
+    }
+}
+impl NetworkError for curl::ErrCode {
+    fn maybe_spurious(&self) -> bool {
+        println!("{}", self.code() as i32);
+        match self.code()  as i32 {
+            7  | //COULDNT_CONNECT
+            5  | //COULDNT_RESOLVE_PROXY
+            6  | //COULDNT_RESOLVE_HOST
+            28 | //OPERATION_TIMEDOUT
+            56 // RECV_ERROR
```

Comment:
curl-sys is a dependency of curl but I cant figure out how to access it. Should I just added it as a dependency?